dev-libs/glib/patches/glib2-2.80.0.patchset

```diff
@@ -0,0 +1,495 @@
+       error('Could not find res_query()')
+     endif
+-- 
+2.50.1
+
+
+From 618f2698c70d7224e63816cc45c81d98e3ad4a5e Mon Sep 17 00:00:00 2001
+From: Gerasim Troeglazov <3dEyes@gmail.com>
+Date: Thu, 5 Jan 2023 18:31:01 +1000
+Subject: Use find_directory for xdgmime
+
+
+diff --git a/gio/xdgmime/xdgmime.c b/gio/xdgmime/xdgmime.c
+index 0909581..338c49c 100644
+--- a/gio/xdgmime/xdgmime.c
++++ b/gio/xdgmime/xdgmime.c
+@@ -29,6 +29,11 @@
+ #include <unistd.h>
+ #include <assert.h>
+ 
++#if defined(__HAIKU__)
++#include <FindDirectory.h>
++#include <fs_info.h>
++#endif
++
+ #ifndef S_ISREG
+ #define S_ISREG(m) (((m) & _S_IFMT) == _S_IFREG)
+ #endif
+@@ -223,6 +228,23 @@ xdg_init_dirs (void)
+ 
+   assert (xdg_dirs == NULL);
+ 
++#ifdef __HAIKU__
++  current_dir = 0;
++  xdg_dirs = calloc (2, sizeof (char *));
++
++  char path[B_PATH_NAME_LENGTH + B_FILE_NAME_LENGTH];
++  dev_t volume = dev_for_path("/boot");
++  if (find_directory(B_SYSTEM_DATA_DIRECTORY, volume, false, path, sizeof(path)) == B_OK)
++    {
++      xdg_data_dirs = strdup(path);
++      strcat (xdg_data_dirs, "/mime/");
++    }
++  else
++    {
++      xdg_data_dirs = strdup("/system/data/mime/");
++    }
++  xdg_dirs[current_dir++] = xdg_data_dirs;
++#else
+   xdg_data_home = getenv ("XDG_DATA_HOME");
+   home = getenv ("HOME");
+   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
+@@ -295,7 +317,7 @@ xdg_init_dirs (void)
+ 
+       ptr = end_ptr;
+     }
+-
++#endif
+   /* NULL terminator */
+   xdg_dirs[current_dir] = NULL;
+ 
+-- 
+2.50.1
+
+
+From 017e09abf4d237406315a3bb7c72061a555f5e06 Mon Sep 17 00:00:00 2001
+From: Alexander von Gluck <alex@terarocket.io>
+Date: Tue, 26 Aug 2025 10:38:29 -0500
+Subject: glib: Disable kqueue on Haiku
+
+
+diff --git a/meson.build b/meson.build
+index 7534542..3b70972 100644
+--- a/meson.build
++++ b/meson.build
+@@ -683,7 +683,6 @@ functions = [
+   'inotify_init1',
+   'issetugid',
+   'kevent',
+-  'kqueue',
+   'lchmod',
+   'lchown',
+   'link',
+@@ -728,6 +727,13 @@ if host_system == 'darwin'
+   functions += ['_NSGetEnviron']
+ endif
+ 
++# Haiku's kqueue implementation is not yet compatible
++if host_system != 'haiku'
++  functions += ['kqueue']
++else
++  have_func_kqueue = false
++endif
++
+ if glib_conf.has('HAVE_SYS_STATVFS_H')
+   functions += ['statvfs']
+ else
+-- 
+2.50.1
+
+
+From 3ba408a5fdaccf149e10e8067ab39892cd31f958 Mon Sep 17 00:00:00 2001
+From: Luc Schrijvers <begasus@gmail.com>
+Date: Wed, 27 Aug 2025 13:05:59 +0200
+Subject: Fixes for mismatching CPU_ZERO, CPU_ISSET and CPU_SET
+
+
+diff --git a/glib/gthread.c b/glib/gthread.c
+index b39acc4..25d23df 100644
+--- a/glib/gthread.c
++++ b/glib/gthread.c
+@@ -1094,6 +1094,12 @@ g_get_num_processors (void)
+   {
+     int idx;
+     int ncores = MIN (sysconf (_SC_NPROCESSORS_ONLN), CPU_SETSIZE);
++#ifdef __HAIKU__
++#define cpu_set_t cpuset_t
++#define CPU_ZERO CPUSET_ZERO
++#define CPU_ISSET CPUSET_ISSET
++#endif
++
+     cpu_set_t cpu_mask;
+     CPU_ZERO (&cpu_mask);
+ 
```

Comment:
In 2.81.0 there is a `CPU_COUNT` change in the lines after these, backporting the variables from this version works, but seems we are missing this one?